File: src/main/java/com/lessmarkup/interfaces/structure/CachedNodeAccess.java
package com.lessmarkup.interfaces.structure;

import com.lessmarkup.dataobjects.NodeAccess;
import java.util.OptionalLong;

public class CachedNodeAccess {
    private NodeAccessType accessType;
    private OptionalLong userId;
    private OptionalLong groupId;
    
    public CachedNodeAccess() {
    }
    
    public CachedNodeAccess(NodeAccess na) {
    }
    
    public NodeAccessType getAccessType() {
        return this.accessType;
    }
    
    public void setAccessType(NodeAccessType accessType) {
        this.accessType = accessType;
    }
    
    public OptionalLong getUserId() {
        return this.userId;
    }
    
    public void setUserId(OptionalLong userId) {
        this.userId = userId;
    }
    
    public OptionalLong getGroupId() {
        return this.groupId;
    }
    
    public void setGroupId(OptionalLong groupId) {
        this.groupId = groupId;
    }
}

File: src/main/scala/com/lessmarkup/interfaces/structure/CachedNodeAccess.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.structure

import com.lessmarkup.dataobjects.NodeAccess
import com.lessmarkup.interfaces.annotations.NodeAccessType

class CachedNodeAccess(nodeAccessType: NodeAccessType, userId: Option[Long], groupId: Option[Long], nodeId: Long) {

  def this(nodeAccess: NodeAccess) {
    this(nodeAccess.accessType, nodeAccess.userId, nodeAccess.groupId, nodeAccess.nodeId)
  }

  def getAccessType: NodeAccessType = nodeAccessType

  def getUserId: Option[Long] = userId

  def getGroupId: Option[Long] = groupId

  def getNodeId = nodeId
}
